aep/general/0162/aep.md.j2

```diff
@@ -1,5 +1,396 @@
 # Resource Revisions
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/16) for more
-information.
+Some APIs need to have resources with a revision history, where users can
+reason about the state of the resource over time. There are several reasons for
+this:
+
+- Users may want to be able to roll back to a previous revision, or diff
+  against a previous revision.
+- An API may create data which is derived in some way from a resource at a
+  given point in time. In these cases, it may be desirable to snapshot the
+  resource for reference later.
+
+**Note:** We use the word _revision_ to refer to a historical reference for a
```

Comment:
I think the issue with "version" is that someone could be referring to V2 of the resource schema, vs V2 of the resource itself. 

I personally don't think there's a cut-and-dry term here - version *could* be ok, but will definitely lead to several conversations where someone has to clarify which concept of "version" they're referring to.

Reply:
If you feel strongly against "revision" and feel strongly for "snapshots", I'm fine with snapshots.

Reply:
> I think the issue with "version" is that someone could be referring to V2 of the resource schema, vs V2 of the resource itself.

I know this is the rationale, but I don't buy that this is something anyone's likely to get confused about in practice.  Maybe I'm just failing to think of a scenario/conversation where there's actual ambiguity?

Re "snapshots"...are there cases where "snapshot" might be used for another pattern, and so it would create the kind of ambiguity that is allegedly a risk with "version"?  I have a specific example in mind (but not sure I can put it in writing somewhere public at the moment).

Reply:
> I know this is the rationale, but I don't buy that this is something anyone's likely to get confused about in practice. Maybe I'm just failing to think of a scenario/conversation where there's actual ambiguity?

For me it's the fact that there'd be conversations around a resource with revisions that may have a v1 / v2 API, and then it gets really confusing (e.g. "version 2 of the item version api" would be a valid sentence).

> Re "snapshots"...are there cases where "snapshot" might be used for another pattern, and so it would create the kind of ambiguity that is allegedly a risk with "version"? I have a specific example in mind (but not sure I can put it in writing somewhere public at the moment).

As is the case with most English words, there's plenty of opportunities for overloaded language and conflicts. FWIW I don't think we have to think much deeper beyond this - We should just decide on nomenclature that's consistent with the AEPs.

One reason to leave it as "revisions" is that aip.dev already defines it that way, and our most likely onboarded to AEPs are those that are using the AIPs already.

Reply:
>  "version 2 of the item version api" would be a valid sentence

Would it?  The API isn't an "item version" API, it's an "item" API where `Item` supports versions.  So "item version API" seems to me like an incorrect name.

I don't think "revision" is terrible, but I think approximately everyone at Roblox has either gotten confused by it, or asked why it's not "version", or just says "version" instead of "revision".  And I don't really want to defend it with "APIs can also be versioned" because I don't find it persuasive.

Reply:
FWIW I'd probably argue that it's really up to someone to argue why an existing precedent *isn't* ok, rather than justify that precedent. Why isn't revision ok? In my experience at Google, I never heard an issue with that naming choice.

Maybe @wora might have an opinion on this as well as a third opinion?

Reply:
after a few offline discussions in the weekly meeting, we've agreed to "revision" - the original terminology.